### DISCUSSION TITLE
Am I really stopping a notebook process when I close the window

### DISCUSSION THREAD

**Discussion by Excespo** [microsoft/vscode-jupyter#9857] [Questions and Answers] [answered]
I use vscode remote and jupyter notebook for my data experiment. And I accidentally notice that when I close the window (of vscode, I clicked "x" to close the window), the process of this notebook still occupy my gpu resource. I think it was what I've done don't actually shut down the process. But it's not so good when I have to free the resource it has occupied. So what can I do to really shut it down?
(p.s. I've tried to use command  `jupyter notebook stop` to stop the server, but the next time I reopen the .ipynb and select the interpreter, it tells me that the kernel cannot be started. And I don't know what has happened. Please explain and give a solution to restart the kernel if possible. Of course the first question is more important to me. Thanks)

**Comment by DonJayamanne** [comment, marked as answer]
> And I accidentally notice that when I close the window (of vscode, I clicked "x" to close the window), the process of this notebook still occupy my gpu resource.

Yes, thats correct and its by design. We never shutdown any remote kernels staretd on another machine.

> But it's not so good when I have to free the resource it has occupied. So what can I do to really shut it down?

Please try using the Powertoys extension found here https://marketplace.visualstudio.com/items?itemName=ms-toolsai.vscode-jupyter-powertoys
Optionally you could use Jupyter itself (classic Jupyter) to manage these sessions
